### ISSUE TITLE
Getting wrong predictions on a model loaded in java vs python using the tensorflow “saved model” api

### ISSUE THREAD

**Issue by JsFlo** [tensorflow/tensorflow#19359]
### System information
- **custom code**:
- **OS Platform and Distribution**: MacOS High Sierra v10.13.3
- **TensorFlow installed from binary**: pip3 install
- **TensorFlow version **: v1.8.0-0-g93bc2e2072 1.8.0
- **Python version**: 3.6.1
- Have I written custom code: Yes
 - Bazel version: n/a
- CUDA/cuDNN version: n/a
- GPU model and memory: n/a
- Exact command to reproduce: In the 'Code' section of this post

### Problem
I'm trying to load a model in Java that was trained in python and saved using the saved model api (`from tensorflow.python.saved_model`).

I'm able to load it in a separate Python script and from Java but the predictions are wrong in the Java version.

I wrote a quick example project with a simple model that demonstrates the "bug" (I'm hoping my misunderstanding).

**Python: OrTraining.py**

Save model after training using Saved Model Api.

    builders = saved_model_builder.SavedModelBuilder(export_path)
    builders.add_meta_graph_and_variables(sess, ["or"], signature_def_map={
        "predict": tf.saved_model.signature_def_utils.predict_signature_def(
            inputs= {"images": x_placeholder},
            outputs= {"scores": hypothesis_function})
        })
    builders.save()

https://github.com/JsFlo/DebuggingSavedModelJava/blob/master/OrTraining.py

**Python: OrLoadSavedModel.py**

Load model in a separate script using the Saved Model Api.

    with tf.Session(graph=tf.Graph()) as sess:
    tf.saved_model.loader.load(sess, ["or"], "orTrainingModels")
    graph = tf.get_default_graph()
    print(graph.get_operations())
    x_placeholder = graph.get_tensor_by_name("or_inputs:0")
    hypothesis_function = graph.get_tensor_by_name("hypothesis_output:0")
    # sess.run("init")
    print(sess.run(hypothesis_function, feed_dict={x_placeholder: np.array([
        np.array([1, 0]),
        np.array([0, 1]),
        np.array([0, 0]),
        np.array([1, 1]),
    ])}))

https://github.com/JsFlo/DebuggingSavedModelJava/blob/master/OrLoadSavedModel.py

**Java: OrLoadSavedModel.java**

Load

     SavedModelBundle savedModelBundle = SavedModelBundle.load("./orTrainingModels", "or");
     Session session = savedModelBundle.session();

Run

    Tensor result = session.runner()
                .feed("or_inputs", tensorInput)
                .fetch("hypothesis_output")
                .run().get(0);

https://github.com/JsFlo/DebuggingSavedModelJava/blob/master/src/main/java/OrLoadSavedModel.java

Both the java version and the python version load and run the graphs without a problem but the java version doens't output the correct predictions.

At first I thought it was because the weights/bias weren't being loaded but I'm able to "run" the weights/bias operation in the java version and see that it has the correct weights that I see in the python script after training.

Check weights in java (https://github.com/JsFlo/DebuggingSavedModelJava)

    Tensor result = session.runner()
                .fetch("da_weights")
                .run().get(0);

### Code
https://github.com/JsFlo/DebuggingSavedModelJava

I run **python3 OrTraining.py** which will "train" (really small model) on my cpu and output a folder called `orTrainingModels` which has the output of the training SavedModel API. (You don't have to run this since the folder is already in the repo but if you did want to then remove the folder first)

    .
    ├── saved_model.pb
    └── variables
        ├── variables.data-00000-of-00001
        └── variables.index

I then run **python3 OrLoadingSaveModel.py** which will output the results of a small (4) validation/test set . I see the correct results in this. 

I run the java version **(OrLoadSavedModel.java)** by opening up the project in Intellij IDEA and clicking play or running **./gradlew clean build :run**

The Java version seems to load the correct weights and bias but the outputs are wrong. Running the java version should output some debugging logs (weights/bias/operations/output).

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by JsFlo** [comment]
This turned out an to be an issue with the way I was feeding the data in. Tensorflow doesn't like to create Tensors of `Boxed Types` (Integer vs int/ Float vs float) and there are checks to see if you're trying to pass in boxed types but it seems like the checks aren't that comprehensive.

*Test from https://github.com/tensorflow/tensorflow/blob/master/tensorflow/java/src/test/java/org/tensorflow/TensorTest.java *

    @Test
    public void testCreateFromArrayOfBoxed() {
        Integer[] vector = new Integer[] {1, 2, 3, 4};
        try (Tensor<Integer> t = Tensor.create(vector, Integer.class)) {
            fail("Tensor.create() should fail because it was given an array of boxed values"); 
        } catch (IllegalArgumentException e) {
         // The expected exception
       }
    }
 
Here's an example of my issue:

        Float[] input = new Float[]{0f, 1f};
        Tensor tensorOutput = Tensor.create(input);
        float[] floatOutput= new float[2];
        tensorOutput.copyTo(floatOutput);
        println(Arrays.toString(floatOutput)); // -7.377E30, -7.377E30


        float[] input = new float[]{0f, 1f};
        Tensor tensorOutput = Tensor.create(input);

        float[] floatOutput= new float[2];
        tensorOutput.copyTo(floatOutput);
        println(Arrays.toString(floatOutput)); // 0, 1
